### PR TITLE
Modify code to prepare HTTPS migration

### PR DESCRIPTION
**Pull request by louisjdmartin**
Following the last meeting here some change that can already be done in DIRAC for HTTPS migration.

It also include a whoami method for services.

BEGINRELEASENOTES

*CS
CHANGE: Separation between threads an logic in Refresher
CHANGE: Separation between threads an logic in ServiceInterface
NEW: Client for Configuration/Server

*Core
NEW: adding new way to detect url in Core.Base.Client (using a kwargs)
NEW: a simple whoami for services

ENDRELEASENOTES

### CODE REVIEW

File: ConfigurationSystem/Client/ConfigurationServerClient.py [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+from DIRAC.Core.Base.Client import Client
+
+
+class ConfigurationServerClient(Client):
+  """
+    Placeholder client to speak with ConfigurationServer.
+  """
+
+  pass
```
Comment (fstagni) [inline ConfigurationSystem/Client/ConfigurationServerClient.py]: Using here
```
self.setServer('Configuration/Server')
```
wouldn't work?

**Comment by fstagni** [comment]
The release notes should mention which DIRAC system is involved.

File: ConfigurationSystem/Client/ConfigurationServerClient.py (hunk 1 above)
Comment (louisjdmartin) [reply ConfigurationSystem/Client/ConfigurationServerClient.py]: Sometime component need to provide an explicit url (for example, the master configuration server choose a slave server to ping it). This line may override the url given from `**kwargs` with a random url from CS.
But I can add this:
```
  def __init__(self, **kwargs):
    Client.__init__(self, **kwargs)
    if not self.getServer():
      self.setServer('Configuration/Server')
```
So by default it use an URL from the CS, but it still possible to give another URL on the keywords arguments.

**Comment by atsareg** [comment]
This PR should be dropped as replcaed by #3815, shouldn't it ?

**Comment by chaen** [comment]
Yes :)

**Pull request closed without merging**